sdk/core/azure-core-test/src/test_proxy_manager.cpp

```diff
@@ -247,6 +247,13 @@ void TestProxyManager::SetProxySanitizer()
   addSanitizer(SanitizerType::Header, storageSasSignatureRegex, "sassig", "x-ms-copy-source");
   addSanitizer(SanitizerType::Header, storageSasSignatureRegex, "sassig", "x-ms-rename-source");
   addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-copy-source-authorization");
+  addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-encryption-key");
+  addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-rename-source");
+  addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-file-rename-source");
+  addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-copy-source");
+  addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-copy-source-authorization");
+  addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-file-rename-source-authorization");
+  addSanitizer(SanitizerType::Header, "(?<auth>.+)", "auth", "x-ms-encryption-key-sha256");
```

Comment:
This will likely break the `CustomerProvidedKey` test which relies on the Sha256 hash'd value matches what the customer provided.

https://github.com/Azure/azure-sdk-for-cpp/blob/abeb4e797b956cc5b44277ca33de37ab1b2f173e/sdk/storage/azure-storage-blobs/test/ut/blob_container_client_test.cpp#L669-L670

Let me follow-up and see if this is necessary to redact. If so, we can evaluate how to opt-out.

Reply:
If the working group deems this needed for redaction, then it must be redacted.

Reply:
Yes, the safest thing to do for now is to redact broadly. So, this becomes an opt-out model rather than opt-in, which is a better default.
We'll need to build a mechanism for test specific opt-outs, where necessary. 

In the meantime, for this specific storage test that might break on re-record, I suggest:
1) Re-evaulate if that particular assert line is really adding much value/is needed, and if not, just remove it.
2) OR if necessary, split the test into a recorded and live portion and move that assert to live-only.